vyper/old_codegen/external_call.py

```diff
@@ -148,10 +148,12 @@ def _external_call_helper(
     sub += arg_packer
 
     if contract_sig.return_type is None:
-        # if we do not expect return data, check that a contract exists at the target address
-        # we can omit this when we _do_ expect return data because we later check `returndatasize`
-        # CMC 20210907 do we need to check this before the call, or can we defer until after?
-        # if we can defer, this code can be pushed down into unpack_returndata
+        # if we do not expect return data, check that a contract exists at the
+        # target address. we must perform this check BEFORE the call because
+        # the contract might selfdestruct. on the other hand we can omit this
+        # when we _do_ expect return data because we later check
+        # `returndatasize` (that check works even if the contract
+        # selfdestructs).
```

Comment:
side note, I don't think you could do a `selfdestruct` and return data at the same time

Reply:
Right. So if a data-returning contract selfdestructs, vyper will revert the transaction anyway because it didn't return any data (thus violating its signature).

Reply:
Exactly. Also I don't think you can (or should be able to) compile a contract that looks like this, lol

Reply:
> Exactly. Also I don't think you can (or should be able to) compile a contract that looks like this, lol

At least in vyper :upside_down_face: